src/int.cr

```diff
@@ -333,6 +333,51 @@ struct Int
     self >> bit & 1
   end
 
+  # Returns the requested range of bits
+  #
+  # ```
+  # 0b10011.bits(0..1) # => 0b11
+  # 0b10011.bits(0..2) # => 0b11
+  # 0b10011.bits(0..3) # => 0b11
+  # 0b10011.bits(0..4) # => 0b10011
+  # 0b10011.bits(0..5) # => 0b10011
+  # 0b10011.bits(1..4) # => 0b1001
+  # ```
+  def bits(range : Range)
```

Comment:
A nillable `bits?` can be considered too.
If done, the return type can be defined to clarify the difference.

Reply:
I'm not sure I understand, when would / should one expect `bits?` return nil?

Reply:
`bits?` would return `nil` instead of an `IndexError`. Not sure it's required though... I'd leave it for now.

Reply:
I think it's wrong to raise `IndexError`. Bits past the bytesize of an int are turned off, imagining like an infinite series of zeros to represent the number. The `bit` method never raises.

Then there's the question about negative indices... in Ruby for some reason it gives you 0 instead of raising. In Crystal it works like that, maybe by chance. Maybe it's not bad to raise on negative indices. But I would not raise on positive indices.

Reply:
That's an interesting point.

I'd definitely raise on negative ones.

Reply:
@asterite it currently doesn't raise on positive indices - it's also tested in the spec
`it { 0b10011_u8.bits(0..16).should eq(0b10011_u8) }` note that we are grabbing `0..16` of `8`

It does raise on negative as it doesn't make sense to me for that to work and the logic to implement would be computationally expensive

Reply:
Ah, you are right. Thanks!